RUFAS/graph_generator.py

```diff
@@ -141,9 +141,11 @@ def generate_graph(
             Generic exception raised by utility functions.
         """
         try:
+            graph_filter_validation_logs = self._validate_graph_filter(graph_details)
             prepared_data, log_pool = self._prepare_plot_data(filtered_pool, graph_details)
+            all_logs = log_pool + graph_filter_validation_logs
 
-            found_errors = any("error" in log for log in log_pool)
+            found_errors = any("error" in log for log in all_logs)
             if found_errors:
                 return log_pool
```

Comment:
```suggestion
                return all_logs
```
Function needs to return `all_logs` at the bottom as well.

Reply:
Right. I thought I'd made that change. Thank you.

I may change this at the end of the function as well in case we end up going with logging warnings for invalid "optional" filters.